lib/floe/workflow/intrinsic_function.rb

```diff
@@ -15,19 +15,20 @@ def self.intrinsic_function?(payload)
 
       def initialize(payload)
         @payload = payload
+        @tree    = Parser.new.parse(payload)
+      rescue Parslet::ParseFailed => err
+        raise Floe::InvalidWorkflowError, err.message
       end
 
       def value(context = {}, input = {})
-        begin
-          tree = Parser.new.parse(payload)
-        rescue Parslet::ParseFailed => err
-          raise Floe::InvalidWorkflowError, err.message
-        end
-
         Floe.logger.debug { "Parsed intrinsic function: #{payload.inspect} => #{tree.inspect}" }
```

Comment:
Should we also move this debug logging to the initialize method?  It's meant to show the parsed tree, so it belongs with parsing.

Reply:
:+1: done